### PR TITLE
Remove paid for from most read container

### PR DESCRIPTION
**Pull request by janua**
## What does this change?

This changes the `update` function in `MostPopularAgent` which has the responsibility of updating an `Agent` periodically. This adds two changes;

 - Use `BrandingFinder` to filer out `paid` content from the agent.
 - Increase the amount requested from Ophan from `10` to `12` - This may not be necessary, but since the container always contains 10, I thought if one gets removed then this will ensure there is still 10 when there is a surging piece of paid/hosted content.

This does not remove sponsorships from the most-viewed container.

I have also fixed the edition that is requested with to the Content API.

## Does this affect other platforms - Amp, Apps, etc?

Web only

@kelvin-chappell @JonNorman

### CODE REVIEW

File: onward/app/feed/MostPopularAgent.scala [hunk 1]
```diff
@@ -59,8 +61,13 @@ class GeoMostPopularAgent(contentApiClient: ContentApiClient, ophanApi: OphanApi
         item: JsValue <- ophanResults.asOpt[JsArray].map(_.value).getOrElse(Nil)
         url <- (item \ "url").asOpt[String]
       } yield {
-        contentApiClient.getResponse(contentApiClient.item(urlToContentPath(url), Edition.defaultEdition))
-          .map(_.content.map(RelatedContentItem(_)))
+        contentApiClient
+          .getResponse(contentApiClient.item(urlToContentPath(url), edition).showTags("tone"))
```
Comment (kelvin-chappell) [inline onward/app/feed/MostPopularAgent.scala]: I think you need section, paid-content tags and isAppropriateForSponsorship field.

**Comment by paperboyo** [comment]
> a surging piece of paid/hosted content

An 🦄? A link or it hasn’t happened 😉

**Comment by janua** [comment]
@kelvin-chappell Thanks!
@paperboyo It has... apparently...! 😉

File: onward/app/feed/MostPopularAgent.scala [hunk 2]
```diff
@@ -59,8 +61,13 @@ class GeoMostPopularAgent(contentApiClient: ContentApiClient, ophanApi: OphanApi
...
+          .map(_.content
+            .filterNot { content =>
+              val branding: Option[Branding] = BrandingFinder.findBranding(content, countryCode)
+              branding.exists(_.isPaid)}
```
Comment (TBonnin) [inline onward/app/feed/MostPopularAgent.scala]: why the intermediary `val` and not `BrandingFinder.findBranding(content, countryCode).exists(_.isPaid)`?

File: onward/app/feed/MostPopularAgent.scala [hunk 3]
```diff
@@ -59,8 +61,17 @@ class GeoMostPopularAgent(contentApiClient: ContentApiClient, ophanApi: OphanApi
...
+        contentApiClient
+          .getResponse(contentApiClient
+            .item(urlToContentPath(url), edition)
+            .showTags("paid-content")
+            .showSection(true)
+            .showFields("isAppropriateForSponsorship"))
+          .map(_.content
+            .filterNot { content =>
+              val branding: Option[Branding] = BrandingFinder.findBranding(content, countryCode)
+              branding.exists(_.isPaid)}
```
Comment (TBonnin) [inline onward/app/feed/MostPopularAgent.scala]: not sure you need the intermediary `val branding` here
```
.map { _.content
    .filterNot { content => BrandingFinder.findBranding(content, countryCode).exists(_.isPaid) }
    .map(RelatedContentItem(_))
}
```

Comment (janua) [inline onward/app/feed/MostPopularAgent.scala]: I think actually we will need it as we will also filter out sponsorship.
` branding.exists(_.isPaid) &&  branding.exists(_.isSponsorship)`

I have to clear this up yet though.

**Comment by janua** [comment]
@TBonnin Turns out, we will let `sponsorship` through on `most-read`.

File: onward/app/feed/MostPopularAgent.scala [hunk 4]
```diff
@@ -59,8 +61,15 @@ class GeoMostPopularAgent(contentApiClient: ContentApiClient, ophanApi: OphanApi
...
         url <- (item \ "url").asOpt[String]
       } yield {
-        contentApiClient.getResponse(contentApiClient.item(urlToContentPath(url), Edition.defaultEdition))
-          .map(_.content.map(RelatedContentItem(_)))
+        contentApiClient
+          .getResponse(contentApiClient
+            .item(urlToContentPath(url), edition)
+            .showTags("paid-content")
+            .showSection(true)
+            .showFields("isAppropriateForSponsorship"))
```
Comment (kelvin-chappell) [inline onward/app/feed/MostPopularAgent.scala]: Actually looking at a capi response, the field is called `isInappropriateForSponsorship`

**Comment by ScottPainterGNM** [comment]
@paperboyo In Australia they massively pushed a piece of GLabs a little while ago and it ended up in the most viewed #truenews

**Comment by janua** [comment]
Thanks @kelvin-chappell, good spot! All patched up now 😄

**Review by kelvin-chappell — APPROVED** [review]
Once appropriate field name is changed 👍

**Pull request merged**